### DISCUSSION TITLE
JKBMS high voltage alarm, support request, how to make a config file

### DISCUSSION THREAD

**Discussion by Fraser86** [Louisvdw/dbus-serialbattery#824] [Q&A]
Please please help. 

I have a 8 cell setup with a jkbms. 
I also have a raspberryPi that is running version V3.0

My trouble is that everyday I am getting notifications that I have a High Voltage alarm despite the OVPR set to 28.4V I get high voltage alarms occurring at 27.78V for example. 
I have come to learn that there is within SerialJkbms a setting within Parameters called Charge Voltage Limit (CVL) and this is set by default to 27.60 volts, in others words 3.45V. This setting is not changeable. 

I am having a problem daily and I have not the experience to know what is happening and how to prevent it. I posted to the forum to try to get some help and answers but was told not to post there but instead to make a support ticket. 
I tried to do this but this requires a knowledge of how to create a config file and to submit that and something called ngrok which I have no knowledge of. I have been trying to understand what this is for several hours over several days but I am stuck.  like to learn but I wanna know what is going on? What is the reason for these annoying High Voltage alarms? How can I prevent this?

@Louisvdw

**Comment by Fraser86** [comment]
I have made an account at Ngrok 
I have manually added the authentication but this next bit is confusing me. I don't understand how I would check the logs . I dont know how this is done and the instructions are not clear to me 



[Config](https://github.com/mr-manuel/venus-os_ngrok#config)

Access to the remote console/GUI → Settings → Services → Ngrok
[Debugging](https://github.com/mr-manuel/venus-os_ngrok#debugging)

The logs can be checked with tail -n 100 -F /data/log/venus-os_ngrok/current | tai64nlocal

You can check, if ngrok is running by executing ps | grep ngrok

If you do not see anything in the logs you can increase the log level in /data/etc/venus-os_ngrok/venus-os_ngrok.py by changing level=logging.INFO to level=logging.DEBUG

**Reply by mr-manuel** [reply]
The debugging you mentioned is for ngrok and not for the dbus-serialbattery. Did you check the dbus-serialbattery [documentation](https://louisvdw.github.io/dbus-serialbattery/)? There is a [troubleshooting](https://louisvdw.github.io/dbus-serialbattery/troubleshoot/) section where you can get all relevant informations.

**Comment by Louisvdw** [comment]
Hi @Fraser86 
Sorry the bug reports are a bit challenging. They are set up for us to be able to get all the information we need to assist you, without asking the same questions to each person that can take a lot of time. You don't need ngrok, just fill out the questions from [support request](https://github.com/Louisvdw/dbus-serialbattery/issues/new?assignees=&labels=support&projects=&template=support_request.yml)

There are settings on the BMS itself (you set that with your phone on the BMS app)
Then there are settings that the driver has. (the config files that you can adjust)
And then there are settings in the GX device itself.

Any one of these could be causing the problem. 
I suspect if your cells are not balanced that this might be causing your issue. Look at the VRM alarms and graphs described [here](https://louisvdw.github.io/dbus-serialbattery/troubleshoot/#alarm-logs-vrm-portal) to check the values of your min/max cell voltages at the time of the alarms.

**Reply by Fraser86** [reply]
I understand it is necessary for the logs and to make it quick to diagnose the problem. 
It's only not clear how to do that. 

when I fill out the support ticket there are two sections that must be filled in (they have *). otherwise I am unable to post. or not?
these two section are these logs. 
But I am away from my system so cannot ssh. 

"There are settings on the BMS itself (you set that with your phone on the BMS app)
Then there are settings that the driver has. (the config files that you can adjust)
And then there are settings in the GX device itself"

indeed I set up the bms through Bluetooth app. with 28.4 as over voltage protection 
my mppts are set up as 28.00V for absorbtion 
27.00V for float. 
I serialbattery jkbms the CVL is 27.6 in parameters 
(which I cannot adjust the value, at least within the venus GUI)
and the "GX device itself". I don't know what you mean. 

I have been checking on my cells through remote console and looks to be balanced.

**Reply by Louisvdw** [reply]
Cells will "look" balanced. You will only know if they are balanced when the battery is full (e.g. if at least one cell is above 3.45V.
This is how a balanced battery will show. When both the min and max cells are close together when the battery is full.
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/26077665/93b261be-5fea-4560-b9e7-f4fab963767b)
You need to check the graph if this is the case for you. If the min cell is around 3.30V and the high cell is around 3.50V then the battery is not balanced and you will get alarms.

See [max cell voltage](https://louisvdw.github.io/dbus-serialbattery/faq/#why-is-the-max-cell-voltage-set-to-345v)
and [high voltage alarm](https://louisvdw.github.io/dbus-serialbattery/faq/#why-do-i-get-a-high-voltage-alarm)
There is also suggestions that you can follow that you can also to remotely.

**Reply by Fraser86** [reply]
Where in Vrm are you able to open this graph. I am searching but have not been able to find Min/Max cell voltages?

**Reply by Fraser86** [reply]
<img width="1267" alt="Screenshot 2023-09-30 at 14 00 10" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/1b04725e-7008-4581-a1fc-c2df08e81019">

**Reply by Louisvdw** [reply]
See the [Advance tab](https://louisvdw.github.io/dbus-serialbattery/troubleshoot/#advanced-section-vrm-portal). Top right is an icon that will drop down all the graph options that you can enable.

**Reply by Fraser86** [reply]
<img width="1011" alt="Screenshot 2023-09-30 at 14 21 49" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/d027164e-b2c6-4cbd-a0bf-c55f2232cd15">

**Reply by Fraser86** [reply]
I found it and switched it on 
this is from a few days ago. 

Max is 3.54
Min is 3.43

**Reply by Fraser86** [reply]
<img width="980" alt="Screenshot 2023-09-30 at 14 26 24" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/9f9dc008-6c69-412b-8e2e-2f428382c5a0">
<img width="1018" alt="Screenshot 2023-09-30 at 14 26 53" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/16f057be-b1da-41c8-80f2-5fc020c080d3">

**Reply by Fraser86** [reply]
@Louisvdw

Checking the min and max voltage, they seem to be ok to me. But I am not sure. Can you see an imbalance that would be create the issue. 
Thanks in advance for the help

**Reply by Louisvdw** [reply]
28 & 29 Sep. your cells are not full, but 27 Sep. is what you want to look at. The min and max is not that far off, but they are both higher than 3.40V meaning all the cells are full. There will always be one cell that is shooting up a bit before the rest, which is normal.
You can also add the BMS charge discharge limits graph. It should then show that the charge current drops when your battery is full and when your cells are going past their max. 

Your BMS takes a bit of time to react, but it does react. But it will still give you the alarm. You can increase the the max cell voltage setting on your BMS a bit, or you can lower your battery charge voltage. This will give your BMS more time to react before the alarm threashold is triggered. 
This is what is described at [max cell voltage](https://louisvdw.github.io/dbus-serialbattery/faq/#why-is-the-max-cell-voltage-set-to-345v)
and [high voltage alarm](https://louisvdw.github.io/dbus-serialbattery/faq/#why-do-i-get-a-high-voltage-alarm)

All in all it does not look like you have any major issues even though you do get the alarm. It is handled and you don't keep on getting the alarm through the day (next charge cycle I assume it will trigger again when the battery is full.

**Reply by Fraser86** [reply]
It happens whenever The SOC reaches 100%, eg full. So it happens sometimes several times a day 
On the 27th of september, there are 7 notifications. 

<img width="843" alt="Screenshot 2023-09-30 at 15 09 22" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/6ce006fc-6b4f-4c71-8e7a-680e06900212">

<img width="991" alt="Screenshot 2023-09-30 at 15 12 22" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/c9fff6ea-12cc-45a5-a09a-36c7b5ee3554">

"or you can lower your battery charge voltage" do you mean lower the Multiplus and Mppt charge value.

**Reply by Louisvdw** [reply]
the settings is done in your GX (e.g. the Remote Console in your Cerbo GX). Just follow the steps in the High Voltage Alarm link I posted above.

**Reply by Fraser86** [reply]
I checked in DVCC and it was switch off. 
I switched it on and the max voltage value is 29.2. 
What value would you recommend, is this not too high 

<img width="796" alt="Screenshot 2023-09-30 at 15 37 07" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/33de8e41-8c3e-4457-a7b3-0eea328dca74">

**Reply by Louisvdw** [reply]
This is what I recommend https://louisvdw.github.io/dbus-serialbattery/faq/#why-is-the-max-cell-voltage-set-to-345v

If DVCC was off you did not follow the install steps. I suggest you review then https://louisvdw.github.io/dbus-serialbattery/general/install

**Reply by Fraser86** [reply]
When I set up, along time ago I did follow the install steps all the way through. but this issue may have caused me to switch it off in the search of a solution. My apologies, but that was by accident. 
The people who made the battery recommended 28.4V and Victrons Lithium Iron Phosphate battery settings are also this. 
By having this off could this have caused the problem. I cannot be sure but I think the issue has been happening from before that.

**Reply by Fraser86** [reply]
my maximum cells voltage should be set to 3.45 x 8 = 27.6. 
I think that I put this higher to avoid the alarm. 
Is this correct?


"3.45V max cell voltage
The cell voltages is used along with the cell count to set the battery voltage (e.g. for 16 cells your battery min voltage will be 3.1 * 16 = 49.6V and max coltage 3.45 * 16 = 55.2V)"

**Reply by Fraser86** [reply]
Im reading this in the documentation but unsure about what its saying 
<img width="735" alt="Screenshot 2023-09-30 at 18 07 01" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/22de39df-f0d4-4902-a9ac-45e22fc1742a">

What is meant my "If your battery works with lower limits"

**Reply by Fraser86** [reply]
for now I have put it like this 
<img width="809" alt="Screenshot 2023-09-30 at 18 11 28" src="https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/abf9c657-29d5-4bb4-b370-e425ddff041f">

**Reply by Fraser86** [reply]
I'm am trying to understand this so the settings are correct. I don't want to keep asking dumb questions. So thanks for your patience. 

So 
Is "if battery works with lower limits" referring to cells with a lower Voltage when full, or something else that I am not getting?
Reading this sentence I guess I'm not understanding lower limit. 

please clarify

**Reply by Fraser86** [reply]
![Screenshot_20231001_093027_Chrome](https://github.com/Louisvdw/dbus-serialbattery/assets/43178840/a5158166-447c-4408-bba5-2a77f517a608)

with some googling I found this.. for educational reason I post it here for other looking for answers

**Reply by Fraser86** [reply]
So I corrected the dvcc setting. 
and  when my battery reached 100% and I again receive high voltage notifications. 

the cell deviation is 0.125

so I again don't understand why I am getting this issue.

**Comment by Fraser86** [comment]
So winter is over and the first days of sun are back and I am again getting this issue. I hoped that the mistake of having dvcc in the wrong position would help but I am still getting High Voltage alarm and battery shutdowns

**Reply by Louisvdw** [reply]
If your battery was not fully charged for a long time then the cells will drift and go out of balance. The Battery Life feature on the Victron system will make sure the battery is reaching full at least once a week which I would suggest you should use.
But if it is out of balance you should slowly charge the battery to give the BMS time to balance the cells again. 

Also remember that this issue is with your battery and not the driver. The driver is only highlighting the issues that you did not see before. 
So it might be worth it to ask your battery supplier to check the hardware. 
You could also lower the charge voltage setting, and also lower the charge current limits. Both will give your BMS more chance to manage the problem cells.

**Reply by Fraser86** [reply]
It's been on Battery Life function throughout the winter. 
How do you change the Charge Voltage setting 
I have looked in parameters - CVL

This is set to 27.6
but I am unable to adjust and not sure if this is what you mean.

**Reply by Louisvdw** [reply]
The parameters CVL is what the driver requests. If you want to override it the best option is in the DVCC settings in your GX device. Just lower that until you do not see the alarm hapenning anymore. 
CVL is is the Charge Voltage and CCL is the Charge Current.
